docs/API-functions.md

```diff
@@ -459,6 +502,126 @@ setting the `SRTO_RENDEZVOUS` option to true, and doing `srt_connect`.
 
 **NOTE:** The port value shall be the same in `local_name` and `remote_name`.
 
+
+Socket group management
+-----------------------
+
+Group types are collected in an `SRT_GROUP_TYPE` type and they are currently:
+
+* `SRT_GTYPE_BROADCAST`: broadcast type, all links are actively used at once
+* `SRT_GTYPE_BACKUP`: backup type, idle links take over connection on disturbance
+* `SRT_GTYPE_BALANCING`: balancing type, share bandwidth usage between links
+
+Functions to be used on groups:
+
+### srt_create_group
+
+```
+SRTSOCKET srt_create_group(SRT_GROUP_TYPE type);
+```
+
+Creates a new group of type `type`. This is typically called on the
+caller side to be next used for connecting to the listener peer side.
+The group ID is of the same domain as socket ID, with the exception that
+the `SRTGROUP_MASK` bit is set on it, unlike for socket ID.
+
+### srt_include
+
+```
+int srt_include(SRTSOCKET socket, SRTSOCKET group);
+```
+
+This function adds a socket to a group. This is only allowed for unmanaged
+groups. No such group type is currently implemented.
+
+### srt_exclude
+
+```
+int srt_exclude(SRTSOCKET socket);
+```
+This function removes a socket from a group to which it currently belongs.
+This is only allowed for unmanaged groups. No such group type is currently
+implemented.
+
+### srt_groupof
+
+```
+SRTSOCKET srt_groupof(SRTSOCKET socket);
+```
+
+Returns the group ID of the socket, or `SRT_INVALID_SOCK` if the socket
+doesn't exist or it's not a member of any group.
+
+### srt_group_data 
+
+```
+int srt_group_data(SRTSOCKET socketgroup, SRT_SOCKGROUPDATA* output, size_t* inoutlen);
+```
+
+* `socketgroup` an existing socket group ID
+* `output` points to an output array
+* `inoutlen` points to a variable set to array's size
+
+This function obtains the current member state of the group specified in
+`socketgroup`. The `output` should point to the array of enough size to get
+all the elements and `inoutlen` to a variable preset to the size of this array.
```

Comment:
all the elements, and `inoutlen` to a variable preset to the size of this array.